superset/db_engine_specs/base.py

```diff
@@ -145,8 +146,80 @@ class BaseEngineSpec:  # pylint: disable=too-many-public-methods
     _date_trunc_functions: Dict[str, str] = {}
     _time_grain_expressions: Dict[Optional[str], str] = {}
     column_type_mappings: Tuple[
-        Tuple[Pattern[str], Union[TypeEngine, Callable[[Match[str]], TypeEngine]]], ...,
-    ] = ()
+        Tuple[
+            Pattern[str],
+            Union[TypeEngine, Callable[[Match[str]], TypeEngine]],
+            GenericDataType,
+        ],
+        ...,
+    ] = (
+        (
+            re.compile(r"^smallint", re.IGNORECASE),
+            types.SMALLINT,
```

Comment:
These should be instantiated: `types.SMALLINT()` instead of `types.SMALLINT`, otherwise the callback logic later on wont work where we check if the type is `callable` (see how Presto implements `VARCHAR`).